From 29f466a69e5ec6ac3253936c9f3f25004f7aff19 Mon Sep 17 00:00:00 2001
From: Liu Ying <b17645@freescale.com>
Date: Mon, 25 Oct 2010 14:01:28 +0800
Subject: [PATCH] ENGR00132882 MXC V4L2 capture:Check slave is present when open dev

This patch makes sure that slave device is present when opening
related video device. It will avoid a kernel dump if slave device
is not present before calling custom open function.

Signed-off-by: Liu Ying <b17645@freescale.com>
---
 drivers/media/video/mxc/capture/mxc_v4l2_capture.c |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/drivers/media/video/mxc/capture/mxc_v4l2_capture.c b/drivers/media/video/mxc/capture/mxc_v4l2_capture.c
index 56cc875..308f949 100644
--- a/drivers/media/video/mxc/capture/mxc_v4l2_capture.c
+++ b/drivers/media/video/mxc/capture/mxc_v4l2_capture.c
@@ -1374,6 +1374,12 @@ static int mxc_v4l_open(struct file *file)
 		return -EBADF;
 	}
 
+	if (cam->sensor == NULL ||
+	    cam->sensor->type != v4l2_int_type_slave) {
+		pr_err("ERROR: v4l2 capture: slave not found!\n");
+		return -EAGAIN;
+	}
+
 	down(&cam->busy_lock);
 	err = 0;
 	if (signal_pending(current))
-- 
1.5.4.4

